### DISCUSSION TITLE
🖋: React Navigationの説明をより分かりやすく

### DISCUSSION THREAD

**Discussion by takei0w0** [ws-4020/mobile-app-crib-notes#137] [Ideas]
学習コンテンツFBより

- useNavigationのカスタムフックの辺りの理解が難しかった
- 公式ドキュメントも読んだが、どのような型が“正しい“のか、もう少し理解を進めなければ実装できないと感じた
- 作ったページをReact Navigationで繋ぐ方法の理解がふわっとしていて、演習問題で焦りました。
- 何回も読みなおして動かしたら何とか理解できた気がする
- ネストされたナビゲーターとModalの説明は混ぜずに分けた方が理解しやすいのではないか
- react-navigationを理解できたとは言えず、もう少し初心者向けの基礎的な説明が欲しい

より基本的な説明＋混乱を招く説明の改善が必要そう

**Comment by ghost** [comment]
> useNavigationのカスタムフックの辺りの理解が難しかった

個人的には、型定義で得られるメリットに比べて実装が難しすぎるので、型定義に関するハンズオンは全部消してしまってよいのではないかと思います。

**Reply by takei0w0** [reply]
全くなくしてしまうと学習の機会が減ってしまうので、ハンズオンからは除いてしまって、オプションとして紹介するくらいはいいのかなと思いました。メリデメも合わせて触れると開発時に採用するかどうか判断できるようになるのでいいかもしれません。

コンテンツのどこに含めるか悩みますが。。

**Reply by ghost** [reply]
型定義の話については、2択かなぁと思ってます。
* 一から十まで解説する
* 全く触れない

オプションとして紹介するとなると、[Type checking with TypeScript | React Navigation](https://reactnavigation.org/docs/typescript)を紹介する感じになると思うんだけど、TypeScriptの型についてかなりの理解が求められると思う。

自分でやってみた感覚としては、自分が作った型定義が正しいことがどこでも保証されないから、「いやこれ結局ランタイムでは何も保証してくれないじゃないか」という印象を受けたんだよね。そういう意味では、「メリデメも合わせて触れると開発時に採用するかどうか判断できるようになる」必要性も薄そうだなぁ、と、、、（Lv.2ってそこまで求められないし、、、）

**Reply by takei0w0** [reply]
> 「いやこれ結局ランタイムでは何も保証してくれないじゃないか」という印象を受けたんだよね。

同じこと思いました。Lv2の到達レベルで判断を求められてないというのもそうですね…。
ここはばっさり触れないようにしましょうか。

**Comment by takei0w0** [comment]
ハンズオンの最初で、TODOアプリの画面遷移を全て作成してもらうことで、画面遷移の実装に集中して学習できるようにしたい。

**Comment by creature-water-valley** [comment]
- [React Navigationの基本](https://ws-4020.github.io/mobile-app-crib-notes/react-native/learn/basic-concepts/react-navigation-basics)で、型定義については無効化していることを明記済み
- [ハンズオンの最初](https://ws-4020.github.io/mobile-app-crib-notes/react-native/learn/todo-app/screens/stack)で、画面遷移を全て作成するように修正済み
